Tests/NIOFileSystemIntegrationTests/FileHandleTests.swift

```diff
@@ -34,7 +34,7 @@ final class FileHandleTests: XCTestCase {
         autoClose: Bool = true,
         _ execute: @Sendable (SystemFileHandle) async throws -> Void
     ) async throws {
-        let path = FilePath("/tmp/\(Self.temporaryFileName())")
+        let path = FilePath("\(FileManager.default.temporaryDirectory.path)/\(Self.temporaryFileName())")
```

Comment:
Can we  use `FileSystem.shared.temporaryDirectory` instead of `FileManager`?

Reply:
Unfortunately not, as that uses a hard-coded `/tmp`, which doesn't exist on Android. I had [originally modified to use Foundation also](https://github.com/apple/swift-nio/commit/dc83c6bc54c9658b63c1dfbc4568c3d0ee41160d#diff-c0a9b635bb72d291323e5d46ff5cebff0a1bd2ed979612d2df50ab90cbef9696R630), but [removed it when asked above](https://github.com/apple/swift-nio/pull/2660#discussion_r1495500012). I figured it was fine here in the tests, since they're already using Foundation, which is why I didn't have to add an import here.

Reply:
So `FileSystem.shared.temporaryDirectory` returns a bogus path on Android, we should fix that instead of resorting on `FileManager` here if possible. How does Foundation get the right tmp path on Android?

Reply:
Can we just default to "/data/local/tmp" on Android then? The comment suggests TMPDIR is rarely set, so this would at least produce a less wrong answer on Android than "/tmp".

Reply:
Done